### PR TITLE
enabled, disabled, in-range, out-of-range, optional, required

### PR DESCRIPTION
**Pull request by NiedziolkaMichal**
Those are interactive examples for pseudo-classes [:enabled](https://developer.mozilla.org/en-US/docs/Web/CSS/:enabled), [:disabled](https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled), [:in-range](https://developer.mozilla.org/en-US/docs/Web/CSS/:in-range), [:out-of-range](https://developer.mozilla.org/en-US/docs/Web/CSS/:out-of-range), [:optional](https://developer.mozilla.org/en-US/docs/Web/CSS/:optional), [:required](https://developer.mozilla.org/en-US/docs/Web/CSS/:required).

image

image

image

image

image

image

### CODE REVIEW

File: live-examples/css-examples/pseudo-class/meta.json [hunk 1]
```diff
@@ -18,6 +18,24 @@
...
             "height": "tabbed-shorter"
         },
+        "disabled": {
+            "cssExampleSrc": "./live-examples/css-examples/pseudo-class/disabled.css",
+            "exampleCode": "./live-examples/css-examples/pseudo-class/disabled.html",
+            "fileName": "pseudo-class-disabled.html",
+            "title": "CSS Demo: :disabled",
+            "type": "tabbed",
+            "defaultTab": "css",
+            "height": "tabbed-shorter"
```
Comment (wbamberg) [inline live-examples/css-examples/pseudo-class/meta.json:64]:
```suggestion
            "height": "tabbed-standard"
```

(the HTML overflows)

File: live-examples/css-examples/pseudo-class/meta.json [hunk 2]
```diff
@@ -18,6 +18,24 @@
...
+        },
+        "enabled": {
+            "cssExampleSrc": "./live-examples/css-examples/pseudo-class/enabled.css",
+            "exampleCode": "./live-examples/css-examples/pseudo-class/enabled.html",
+            "fileName": "pseudo-class-enabled.html",
+            "title": "CSS Demo: :enabled",
+            "type": "tabbed",
+            "defaultTab": "css",
+            "height": "tabbed-shorter"
```
Comment (wbamberg) [inline live-examples/css-examples/pseudo-class/meta.json]:
```suggestion
            "height": "tabbed-standard"
```

(the HTML overflows)

File: live-examples/css-examples/pseudo-class/meta.json [hunk 3]
```diff
@@ -153,6 +180,33 @@
...
+        },
+        "required": {
+            "cssExampleSrc": "./live-examples/css-examples/pseudo-class/required.css",
+            "exampleCode": "./live-examples/css-examples/pseudo-class/required.html",
+            "fileName": "pseudo-class-required.html",
+            "title": "CSS Demo: :required",
+            "type": "tabbed",
+            "defaultTab": "css",
+            "height": "tabbed-shorter"
```
Comment (wbamberg) [inline live-examples/css-examples/pseudo-class/meta.json]:
```suggestion
            "height": "tabbed-standard"
```

(the HTML overflows)

File: live-examples/css-examples/pseudo-class/meta.json [hunk 4]
```diff
@@ -153,6 +180,33 @@
...
             "height": "tabbed-shorter"
         },
+        "optional": {
+            "cssExampleSrc": "./live-examples/css-examples/pseudo-class/optional.css",
+            "exampleCode": "./live-examples/css-examples/pseudo-class/optional.html",
+            "fileName": "pseudo-class-optional.html",
+            "title": "CSS Demo: :optional",
+            "type": "tabbed",
+            "defaultTab": "css",
+            "height": "tabbed-shorter"
```
Comment (wbamberg) [inline live-examples/css-examples/pseudo-class/meta.json]:
```suggestion
            "height": "tabbed-standard"
```

(the HTML overflows)

File: live-examples/css-examples/pseudo-class/disabled.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+label {
+    display: block;
+    margin-top: 1em;
+}
+
+*:disabled {
+    background-color: indianred;
```
Comment (wbamberg) [inline live-examples/css-examples/pseudo-class/disabled.css]: This makes the text really hard to read. Choose a different colour perhaps?

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thank you @NiedziolkaMichal ! Most of my comments are to make sure the editor is tall enough to fit the HTML. And one contrast issue.

**Comment by NiedziolkaMichal** [comment]
I have updated colors in example disabled. Is that ok?
image

**Comment by NiedziolkaMichal** [comment]
I also added html attribute [disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled).

image

**Review by wbamberg — APPROVED** [review]
👍 thank you!

**Comment by wbamberg** [comment]
CI failure was not you, it was caused by https://github.com/mdn/interactive-examples/commit/c6b5dc61c6b558c58091b8d974e752e64fc040d2 - I'll make a separate fix.

**Pull request merged**